libpdbg/device.c

```diff
@@ -657,6 +657,67 @@ static void dt_link_virtual(struct pdbg_target *node, struct pdbg_target *vnode)
 	vnode->vnode = node;
 }
 
+/**
+ * @brief A function to delete the children from the list
+ * 
+ * This function is called for a valid target/node to delete the children
+ * from the list. Should be only called internally. Recursive in nature
+ * 
+ * @param target The target node for which the children will be reset
+ * 
+ * @see pdbg_reset_dt_root for more details
+ */
+static void pdbg_reset_children(struct pdbg_target* target)
+{
+	struct pdbg_target* childTraget = NULL;
+	enum pdbg_target_status status;
+
+	/* Does this target actually exist? */
+	if (target)
+	{
+		status = pdbg_target_status(target);
+		if (status != PDBG_TARGET_ENABLED)
+			return;
+	}
+	pdbg_for_each_child_target(target, childTraget)
+	{			
+		if (childTraget)
+		{
+			pdbg_reset_children(childTraget);
+		}
+		if (!list_empty(&target->children) && target->class_link.next && target->class_link.prev)
+		{
+			list_del_from(&target->children, &target->class_link);
+		}
+	}
+	if (childTraget)
+	{
+		free(childTraget);
+		childTraget = NULL;
+	}
```

Comment:
should this be inside the for loop, can we ensure that how many targets were present and if we called that many free(childTarget) or else we will be having memory leak issues.

Reply:
I too observed it. But it looks like a recursive method and will come here for every child. May be a comment could help